ledger-service/http-json-testing/src/main/scala/com/daml/http/HttpServiceUserFixture.scala

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.Uri
+import akka.http.scaladsl.model.headers.Authorization
+import akka.stream.Materializer
+import com.daml.grpc.adapter.{AkkaExecutionSequencerPool, ExecutionSequencerFactory}
+import com.daml.http.HttpServiceTestFixture.{authorizationHeader, postRequest}
+import com.daml.http.util.ClientUtil.uniqueId
+import com.daml.jwt.JwtSigner
+import com.daml.jwt.domain.{Jwt, DecodedJwt}
+import com.daml.platform.sandbox.SandboxRequiringAuthorizationFuns
+import scalaz.syntax.show._
+import scalaz.syntax.tag._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait HttpServiceUserFixture {
+  protected def testId: String
+
+  // TODO(SC) we make asys, mat, and aesf "win" to match the prior behavior when
+  // these defns were in AbstractHttpServiceIntegrationTestFuns.  But it
+  // might be possible to phase them out entirely in favor of AkkaBeforeAndAfterAll
+  import shapeless.tag, tag.@@ // used for subtyping to make `AHS ec` beat executionContext
+  implicit val `AHS asys`: ActorSystem @@ this.type = tag[this.type] {
+    import com.typesafe.config.ConfigFactory
+    val customConf = ConfigFactory.parseString("""
+      akka.http.server.request-timeout = 60s
+    """)
+    ActorSystem(testId, ConfigFactory.load(customConf))
+  }
+  implicit val `AHS mat`: Materializer @@ this.type = tag[this.type](Materializer(`AHS asys`))
+  implicit val `AHS aesf`: ExecutionSequencerFactory @@ this.type =
+    tag[this.type](new AkkaExecutionSequencerPool(testId)(`AHS asys`))
+  // XXX(SC) `AHS ec` has beat executionContext for much longer
+  implicit val `AHS ec`: ExecutionContext @@ this.type = tag[this.type](`AHS asys`.dispatcher)
+
+  def jwtForParties(uri: Uri)(
+      actAs: List[String],
+      readAs: List[String],
+      ledgerId: String,
+      withoutNamespace: Boolean = false,
+      admin: Boolean = false,
+  )(implicit ec: ExecutionContext): Future[Jwt]
+
+  protected val jwtAdminNoParty: Jwt
```

Comment:
Minor, I prefer def in trait

Reply:
Is there a specific reason for that?

Reply:
The implementation can decide val or not. It's rather specific to constrain this in the trait, and likely not needed.

Also vals in trait have their issues, (initialization order) not as is done here, but in general, so I prefer to avoid it, so you simply can't make the mistake.

Reply:
if the initialization is an issue I would go for a lazy val

Reply:
I would prefer def in trait since it is most abstract way of just saying you need to define something. Implementer can decide what they want to do and why. If you want to use lazy val in the implementation, sure.

Reply:
I prefer abstract def unless having a stable path is necessary or convenient. 0907358132926218c8ad414030b2c41780d8d0a7